Source/DafnyCore/DafnyFile.cs

```diff
@@ -23,6 +24,13 @@ public class DafnyFile {
 
   public static DafnyFile CreateAndValidate(ErrorReporter reporter, IFileSystem fileSystem,
     DafnyOptions options, Uri uri, IToken origin) {
+
+    if (uri.Scheme == "doo") {
```

Comment:
VSCode only allows you to have custom handlers for custom schemes, which seems fair to me, but it means we have to introduce a `doo:` scheme, which is at odds with that we already handle the .doo extension as being special, even if no `doo:` scheme is used. That's why the code looks a bit funky now.

If we were more strict, only allowing doo files being used as libraries, and only allowing dfy files being used as sources, then we wouldn't need to rely on the extension checking, but that seems like something to consider for the future.